pdfminer/high_level.py

```diff
@@ -3,22 +3,34 @@
 import logging
 import sys
 from io import StringIO
+from typing import Any, BinaryIO, Container, Iterator, Optional, cast
 
 from .converter import XMLConverter, HTMLConverter, TextConverter, \
     PDFPageAggregator
 from .image import ImageWriter
-from .layout import LAParams
-from .pdfdevice import TagExtractor
+from .layout import LAParams, LTPage
+from .pdfdevice import PDFDevice, TagExtractor
 from .pdfinterp import PDFResourceManager, PDFPageInterpreter
 from .pdfpage import PDFPage
-from .utils import open_filename
-
-
-def extract_text_to_fp(inf, outfp, output_type='text', codec='utf-8',
-                       laparams=None, maxpages=0, page_numbers=None,
-                       password="", scale=1.0, rotation=0, layoutmode='normal',
-                       output_dir=None, strip_control=False, debug=False,
-                       disable_caching=False, **kwargs):
+from .utils import open_filename, FileOrName, AnyIO
+
+
+def extract_text_to_fp(inf: BinaryIO,
+                       outfp: AnyIO,
+                       output_type: str = 'text',
+                       codec: str = 'utf-8',
+                       laparams: Optional[LAParams] = None,
+                       maxpages: int = 0,
+                       page_numbers: Optional[Container[int]] = None,
+                       password: str = "",
+                       scale: float = 1.0,
+                       rotation: int = 0,
+                       layoutmode: str = 'normal',
+                       output_dir: Optional[str] = None,
+                       strip_control: bool = False,
+                       debug: bool = False,
+                       disable_caching: bool = False,
```

Comment:
When reformatting functions, can you do it like this: 

```python
def extract_text_to_fp(
    inf: BinaryIO,
    outf: AnyIO,
    ...
    **kwargs: Any,
) -> None:
```

I've read a blog once that summarized the benefits of this approach. Can't find it anymore, but one big benefit is that git diffs will perfectly match the arguments that actually changed.